v-next/hardhat/src/internal/builtin-plugins/console/task-action.ts

```diff
@@ -0,0 +1,46 @@
+import type { NewTaskActionFunction } from "@ignored/hardhat-vnext-core/types/tasks";
+import type { REPLServer } from "node:repl";
+
+import path from "node:path";
+import repl from "node:repl";
+
+import { getCacheDir } from "@ignored/hardhat-vnext-core/global-dir";
+import debug from "debug";
+
+const log = debug("hardhat:core:tasks:console");
+
+interface ConsoleActionArguments {
+  commands: string[];
+}
+
+const consoleAction: NewTaskActionFunction<ConsoleActionArguments> = async (
+  { commands },
+  _hre,
+) => {
+  const globalCacheDir = await getCacheDir();
+
+  return new Promise<REPLServer>((resolve) => {
+    // Start a new REPL server with the default options
+    const replServer = repl.start();
+
```

Comment:
I think what's missing here is exporting the `hre`. I believe that's done with `replServer.context.hre = hre`.

Reply:
We should also consider exposing each `hre` field in the context, similar to what we do in `v-next/hardhat/src/index.ts`